### PR TITLE
dataloader: create new tables and populate with sample_data from csv

### LINKED ISSUE #7: models: methods for programmatically loading data into h2 using script
### Background
- h2 is designed to run the database in active memory, meaning that it is not persisted and needs to be created each time
- we already have an ample supply of data in the [sample csv files](https://drive.google.com/drive/u/0/folders/1NC-R__Wryf0z3VfcH_z_vHhryeYiHTpT)
- existing work to write the `create table` sql statements is captured in the [Table Creation Commands document](https://docs.google.com/document/d/1sHFz-0iTkJPo6KOA85LeRdkvD45alrZaIT_lQ38dJrg/edit)

### Success Criteria
- separate methods exist to generate each table:
- tables are populated with the sample data (with relations taken into account)
- this is run immediately after the connection is established, and is adequately logged

### PR DESCRIPTION
**Pull request by atla5**
Resolves #7 

### Description of changes
- [x] manually create new tables via H2 console 
- [x] create new `DataLoader` with main method for loading data 
- [x] add `insertNew*` helpers to each `*Proxy` class.
- [x] use the existing controller connection to update the tables through the code (instead of manually)
- [x] update `vendors` table to add in the vendor name.
- [ ] ~remove all `date`s and customer demographics from db~ (moved to a separate PR)

_note: table creation itself was done manually in H2, and will need to be updated once schema and sample data are all confirmed with the team_

### CODE REVIEW

File: src/main/java/com/rainforestcommerce/rcdb/controllers/DataLoader.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class DataLoader {
+    public static String GENERIC_INSERT_STATEMENT = "INSERT into %s VALUES %s;";
+
+    private static String dataDirectory = "/Users/aidan/projects/school/csci-320/src/main/resources/sample_data";
```
Comment (atla5) [inline src/main/java/com/rainforestcommerce/rcdb/controllers/DataLoader.java]: need to make this generic. 👍

File: src/main/java/com/rainforestcommerce/rcdb/controllers/DataLoader.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        loadCustomers();
+        loadStores();
+        loadInventory();
+        loadPurchases(); // store and product purchases
+        System.exit(0);
+    }
+
+    public static boolean insertValuesIntoTable(String values, String tableName){
+        String insertCommand = String.format(GENERIC_INSERT_STATEMENT, tableName, values);
+        System.out.println(insertCommand);
```
Comment (atla5) [inline src/main/java/com/rainforestcommerce/rcdb/controllers/DataLoader.java]: switch to `LOGGER` and set to `debug`

**Comment by grahamhome** [comment]
Resolved conflicts with master. @atla5 I cannot approve this PR yet as I cannot run it - where is the code to create the tables?

**Comment by atla5** [comment]
the `create_tables.sql` ticket is being created in #18, so it should be approved without it.

(i'd answered your question quite a while ago on slack, but figured i should add it here too for provenance)

**Review by aaa1008 — APPROVED** [review]
Looks good to me.

**Pull request merged**